src/app/Http/Controllers/ApiController.php

```diff
@@ -126,12 +129,79 @@ public function updateAvailability(Request $request, Location $location) {
     public function armorvax(Request $request) {
         $data = $request->all();
         $data_string = json_encode($data, JSON_PRETTY_PRINT);
-        $path = sprintf('armorvax/%s/u%d_%s.json', config('app.env'), $request->user()->id, date('Y-m-d_his'));
+        $data_collection = collect(json_decode($data_string));
 
-        // Write the full JSON data to a file on S3
-        \Storage::disk('s3')->write($path, $data_string);
+        /**
+         * First things first: delete all the existing availabilities
+         * associated with the ArmorVax sites.
+         */
+        Location::locationsWithSource('ArmorVax')->each(
```

Comment:
Combined with the scopeBySource change below, you would change this to `Location::bySource('ArmorVax')`.